sql/analyzer/analyzer.go

```diff
@@ -407,9 +430,13 @@ func NewFinalizeNestedSubquerySel(sel RuleSelector) RuleSelector {
 func NewFinalizeSubqueryExprSelector(sel RuleSelector) RuleSelector {
 	return func(id RuleId) bool {
 		switch id {
-		case finalizeSubqueriesId:
-			// Don't run finalizeSubqueries on subqueries, since calling it on the root of the statement will
-			// recursively handle subqueries from the bottom of the plan up.
+		case
+			// skip recursive resolve rules
+			resolveSubqueryExprsId,
```

Comment:
(minor) similar comment for `resolveSubqueryExprs`... I needed to combine the handling of subqs and subq expressions when I added outer scope visibility to sqas so the scopes would be nested correctly, so the `resolveSubqueryExprs` rule has been removed. Sorry for missing cleaning up these Ids!